### PR TITLE
Add per world chunks

### PR DESCRIPTION
**Pull request by sinkillerj**
This pull adds a new event which mods can subscribe to which is only fired on initial world creation, this gives them a place to register per world chunks and allows them to be randomized by the mod in question. The requested chunks are saved to a file so that even if they do not generate during the game session the server still knows to later.

### CODE REVIEW

**Comment by sinkillerj** [comment]
I will try to get a example mod up soon for other devs to look at.

**Comment by sinkillerj** [comment]
Usage info has been added, this should now be good to go.

File: src/main/java/lilylicious/staticchunkmanager/chunk/ChunkRequest.java [hunk 1]
```diff
@@ -46,6 +47,15 @@ public ChunkRequest(InputStream is, int x, int y, boolean... extras) {
         this.matchBiomes = extras.length > 0 ? extras[0] : false;
     }
 
+    public ChunkRequest(InputStream is, String path, int x, int y, boolean... extras) {
```
Comment (Lilylicious) [inline src/main/java/lilylicious/staticchunkmanager/chunk/ChunkRequest.java]: Adjust this in accordance with https://github.com/Lilylicious/StaticChunkManager/blob/master/src/main/java/lilylicious/staticchunkmanager/chunk/ChunkRequest.java#L39

Other than that, it looks fine

**Comment by sinkillerj** [comment]
This should be good now.

**Pull request merged**